moveit_core/collision_detection/include/moveit/collision_detection/collision_common.h

```diff
@@ -166,25 +161,8 @@ struct CollisionResult
     cost_sources.clear();
   }
 
-  /** \brief Throttled warning of the first collision pair, if any. Other collisions are logged at the DEBUG level */
-  void print()
-  {
-    if (!contacts.empty())
-    {
-      ROS_WARN_STREAM_THROTTLE_NAMED(collision_log_period, collision_log_name,
-                                     "Objects in collision (printing first collision pair of "
-                                         << contacts.size() << "): " << contacts.begin()->first.first << ", "
-                                         << contacts.begin()->first.second);
-
-      // Log all collisions at the debug level
-      ROS_DEBUG_STREAM_THROTTLE_NAMED(collision_log_period, collision_log_name, "Objects in collision:");
-      for (auto contact : contacts)
-      {
-        ROS_DEBUG_STREAM_THROTTLE_NAMED(collision_log_period, collision_log_name,
-                                        "\t" << contact.first.first << ", " << contact.first.second);
-      }
-    }
-  }
+  /** \brief Throttled warning printing the first collision pair, if any. All collisions are logged at DEBUG level */
+  void print() const;
```

Comment:
Moving the `print()` method into a .cpp file resolves the scope issue mentioned in https://github.com/ros-planning/moveit/pull/2275#discussion_r475567741.
As `print()` isn't lightweight nor performance-critical, it shouldn't be inline anyway.

Reply:
That sounds good to me. I was thinking about doing that but waiting for a nudge from a reviewer.